maps/randomvaults/mining/mining_surprise_items.dm

```diff
@@ -59,4 +67,85 @@
 /datum/component/ai/hearing/say_response/dusky_when/initialize()
 	if(..())
 		hear_args = list("My production serial seems to be dated to roughly [rand(2300,2399)], ain't seen a customer in about [rand(12,120)] years.")
-		return TRUE
\ No newline at end of file
+		return TRUE
+
+/datum/component/ai/hearing/order/bardrinks/select_reagents/dusky
+	notfoundmessages = list("Sorry pal, I don't think I recognise that kinda thing.",
+							"Don't think we got a recipe for that back in storage, gotta try again.",
+							"I don't know what kinda crazy stuff they're serving on yer planet but we dont have that, try somethin' else.")
+	freemessages = list("For you, on the house, fella!",
+						"I s'pose I could fix up a free servin' for ya, just cus I like ya and all.",
+						"Y'know, today I'm feelin extra generous, so ya get this one free!")
+	toomuchmessages = list("Easy pal, I can only hold so many glasses in my ol' compartments. Try gettin' your stuff first and then orderin' again.",
+							"Look, I ain't made to carry the world on my shoulders here, get your stuff before orderin' again.",
+							"Sorry, inventory's full, come get yer stuff first.")
+	acceptable_recipe_reagents = list(RADIUM)
+	var/lastwrongitemtime
+
+/datum/component/ai/hearing/order/bardrinks/select_reagents/dusky/process()
+	..()
+	if(!items2deliver.len && isliving(parent))
+		var/mob/living/M=parent
+		if(!M.isDead())
+			var/amount = 0
+			var/list/sheets = list()
+			var/list/containers = list()
+			var/turf/checkloc = get_step(M,M.dir)
+			for(var/obj/item/stack/sheet/mineral/plasma/P in checkloc)
+				amount += P.amount
+				sheets += P
+				if(amount >= 50)
+					break
+			if(amount < 50)
+				for(var/obj/item/weapon/reagent_containers/RC in checkloc)
+					amount += RC.reagents.get_reagent_amount(PLASMA)
+					containers += RC
+					if(amount >= 50)
+						break
+			if(amount)
+				M.say(pick("Well how about that, you found some plasma for me to fix up. Let's see if I can make somethin' for ya.",
+						"Aw, my favorite, I love brewin' with plasma. I think I'll make something here with it.",
+						"You found me some plasma? Well thank ya, here's a little something..."))
+				playsound(M.loc, pick('sound/items/polaroid1.ogg','sound/items/polaroid2.ogg'), 70, 1)
+				for(var/obj/O in sheets)
+					qdel(O)
+				for(var/obj/O2 in containers)
+					O2.reagents.del_reagent(PLASMA)
+				for(var/i in 1 to round(clamp(amount/10,1,5)))
+					// i could write something that finds every bar drink that has plasma in its recipe but let's face it,
+					// it's just this stuff and it's less expensive to look for these
```

Comment:
while true, if anyone adds another plasma drink they'll also need to update this, and with no hint on the recipes page that this is the case, you'd have a disconnect between the two